Composer/packages/client/src/pages/design/DesignPage.tsx

```diff
@@ -311,6 +310,30 @@ const DesignPage: React.FC<RouteComponentProps<{ dialogId: string; projectId: st
     }
   }, [location]);
 
+  // get newest label for breadcrumbs
+  useEffect(() => {
+    if (currentDialog.content) {
+      setBreadcrumbs((prev) => {
+        return prev.map((b) => {
+          const separatorIndex = b.key.indexOf('-');
+          const type = b.key.substr(0, separatorIndex);
+          const name = b.key.substr(separatorIndex + 1);
```

Comment:
You can use a Regexp here - try something like
```
const regexp = /(?<type>[a-z]+)-(?<name>.*)/;
const { type, name } = match?.groups;
```

Since the `type` is something we have full control over, we know the match will succeed, but there might be some additional guards that need to be there so TS doesn't complain.

Reply:
I have checked the `type` variable. It is just set & use in the design page, we can add new type if have additional guards. I  create a const variable `types` to manager the type. 
```
const Types = {
  Dialog: 'dialog',
  Trigger: 'trigger',
  Action: 'action',
};
```